packages/medusa/src/services/payment-collection.ts

```diff
@@ -0,0 +1,162 @@
+import { DeepPartial, EntityManager, IsNull } from "typeorm"
+import { MedusaError } from "medusa-core-utils"
+
+import { FindConfig } from "../types/common"
+import { buildQuery, isDefined } from "../utils"
+import { PaymentCollectionRepository } from "../repositories/payment-collection"
+import { PaymentCollection, PaymentCollectionStatus } from "../models"
+import { TransactionBaseService } from "../interfaces"
+import { EventBusService } from "./index"
+
+import { CreatePaymentCollectionInput } from "../types/payment-collection"
+
+type InjectedDependencies = {
+  manager: EntityManager
+  paymentCollectionRepository: typeof PaymentCollectionRepository
+  eventBusService: EventBusService
+}
+
+export default class PaymentCollectionService extends TransactionBaseService {
+  static readonly Events = {
+    CREATED: "payment-collection.created",
+    UPDATED: "payment-collection.updated",
+    DELETED: "payment-collection.deleted",
+  }
+
+  protected readonly manager_: EntityManager
+  protected transactionManager_: EntityManager | undefined
+  protected readonly eventBusService_: EventBusService
+  // eslint-disable-next-line max-len
+  protected readonly paymentCollectionRepository_: typeof PaymentCollectionRepository
+
+  constructor({
+    manager,
+    paymentCollectionRepository,
+    eventBusService,
+  }: InjectedDependencies) {
+    // eslint-disable-next-line prefer-rest-params
+    super(arguments[0])
+
+    this.manager_ = manager
+    this.paymentCollectionRepository_ = paymentCollectionRepository
+    this.eventBusService_ = eventBusService
+  }
+
+  async retrieve(
+    paymentCollectionId: string,
+    config: FindConfig<PaymentCollection> = {}
+  ): Promise<PaymentCollection> {
+    const manager = this.transactionManager_ ?? this.manager_
+    const paymentCollectionRepository = manager.getCustomRepository(
+      this.paymentCollectionRepository_
+    )
+
+    const query = buildQuery({ id: paymentCollectionId }, config)
+    const paymentCollection = await paymentCollectionRepository.findOne(query)
+
+    if (!paymentCollection) {
+      throw new MedusaError(
+        MedusaError.Types.NOT_FOUND,
+        `Payment collection with id ${paymentCollectionId} was not found`
+      )
+    }
+
+    return paymentCollection
+  }
+
+  async create(
+    data: CreatePaymentCollectionInput,
+    context: { loggedInUserId: string }
```

Comment:
thought: Would it make sense to rename to `createdBy` to make the field more generic to the service consumer as the column is not constrained to a user id? E.g. let's say an automation triggers the creation of a PaymentCollection, the consumer might want to have `"created_by": "some_automation"` here.

Alternatively, could we move it to `CreatePaymentCollectionInput` as it is a required prop anyways? And then in our route populate it with the logged in user.

Let me know what you think.

Reply:
Yeah, I like the idea of making it more generic.